diff --git a/ThirdParty/png/vtkpng/pngpriv.h b/ThirdParty/png/vtkpng/pngpriv.h
index 3581f6791999d085fe54d0a65f87e15918c908cd..cf23f68c4ad47929de6d62d9f241ad37b034a42d 100644
--- a/ThirdParty/png/vtkpng/pngpriv.h
+++ b/ThirdParty/png/vtkpng/pngpriv.h
@@ -88,6 +88,11 @@
 #  endif
 #endif
 
+#define PNG_ARM_NEON_OPT 0
+#define PNG_MIPS_MSA_OPT 0
+#define PNG_POWERPC_VSX_OPT 0
+#define PNG_INTEL_SSE_OPT 0
+
 /* Compile time options.
  * =====================
  * In a multi-arch build the compiler may compile the code several times for the
