diff --git a/Rendering/Qt/vtkQtLabelRenderStrategy.cxx b/Rendering/Qt/vtkQtLabelRenderStrategy.cxx
index eca65c64cd..4125d579ac 100644
--- a/Rendering/Qt/vtkQtLabelRenderStrategy.cxx
+++ b/Rendering/Qt/vtkQtLabelRenderStrategy.cxx
@@ -46,6 +46,7 @@
 #include <QPixmap>
 #include <QTextDocument>
 #include <QTextStream>
+#include <QPainterPath>
 
 vtkStandardNewMacro(vtkQtLabelRenderStrategy);
 
diff --git a/Rendering/Qt/vtkQtStringToImage.cxx b/Rendering/Qt/vtkQtStringToImage.cxx
index fbb9b78f05..88181001e6 100644
--- a/Rendering/Qt/vtkQtStringToImage.cxx
+++ b/Rendering/Qt/vtkQtStringToImage.cxx
@@ -35,6 +35,7 @@
 #include <QString>
 #include <QTextDocument>
 #include <QTextStream>
+#include <QPainterPath>
 
 namespace
 {
